### DISCUSSION TITLE
Can hasura-engine read-replicas all share a single metadata database?

### DISCUSSION THREAD

**Discussion by larsblumberg** [hasura/graphql-engine#8946] [Question] [answered]
**Context:**

Without using Hasura Cloud to still achieve postgres replica load-balancing, it's possible to run a separate, second `hasura-engine` read-only instance which connects to a postgres read-replica instance. The main/master `hasura-engine` instance connects against postgres master for R/W ops while the hasura read-only instance connects against the postgres read-replica in read-only mode.

To have `hasura-engine` connect in read-only mode, one needs to strip off cron triggers and event triggers from the metadata. Notice that otherwise hasura will connect in read-write mode which will be rejected by the postgres read-replica.

Doing so you can load-balance graphql read-only operations such as live-subscriptions and queries against the hasura master and the single hasura read-replica. 

**Problem:**

Above setup works well and now I'd like to increase the postgres read-replica to `2` read-only instances for client apps to load-balance against.

**Questions:**

Can `hasura-engine` read-only instances all share a single hasura metadata database? Or does each hasura instance needs its very own metadata database so that `number of hasura read-only instances` = `number of metadata instances`?

**Comment by adas98012** [comment, marked as answer]
As per the Hasura [Docs](https://hasura.io/docs/latest/databases/connect-db/read-replicas/):

Hasura Cloud and Hasura Enterprise can load balance queries and subscriptions across read replicas while sending all mutations and **_metadata API calls to the primary._**

So, it will be a single metadata database (attached to primary) and will be shared across the Read Replicas.

**Reply by larsblumberg** [reply]
Thanks, that answers my question concerning all read-replicas being able to share the same metadata postgres instance.
